content/doc/book/pipeline/scaling-pipeline.adoc

```diff
@@ -0,0 +1,88 @@
+
+== Requirements To Use Durability Settings
+
+* Jenkins LTS 2.73+ or higher (or a weekly 2.62+)
+* Current Pipeline Plugin versions from near the end of January 2018 for all the following:
+    - Pipeline: API (workflow-api)
+    - Pipeline: Groovy (workflow-cps)
+    - Pipeline: Job (workflow-job)
+    - Pipeline: Supporting APIs (workflow-support)
+    - Pipeline: Multibranch (workflow-multibranch) - optional, only needed to enable this setting for multibranch pipelines.
+* Restart the master to use the updated plugins - note: you need all of them to take advantage.
+
+
+== What Are The Durability Settings? 
+
+* Performance optimized mode ("PERFORMANCE_OPTIMIZED") - *Greatly* reduces disk I/O.  If Pipelines do not finish AND Jenkins is not shut down gracefully, they may lose data and behave like freestyle projects -- see details above.
+
+* Maximum durability ("MAX_SURVIVABILITY") - behaves just like Pipeline did before, slowest option.  Use this for running your most critical Pipelines.
+
+* Less durable, a bit faster ("SURVIVABLE_NONATOMIC") - Writes data with every step but avoids atomic writes. This is faster than maximum durability mode, especially on networked filesystems.  It carries a small extra risk (details above under "What Am I Giving Up").
+
+== Suggested Best Practices And Tips for Durability Settings
+
+* Use the PERFORMANCE_OPTIMIZED mode for most pipelines and especially basic build-test Pipelines or anything that can simply be run again if needed.
+* Use MAX_SURVIVABILITY and higher durability modes for Pipelines where you need a guaranteed record of their execution (auditing).
+* Use MAX_SURVIVABILITY and other higher durability modes when pipelines modify the state of critical infrastructure. For example, use it for production deployments.
+* Set a global default (see above) of PERFORMANCE_OPTIMIZED for the Durability Setting, and then where needed set MAX_SURVIVABILITY on specific Pipeline jobs or Multibranch Pipeline branches ("master" or release branches).
+* You can force a Pipeline to persist data by pausing it. 
+* 
+
+= Other Scaling Suggestions
+
+* *Whenever possible, run Jenkins with fast SSD-backed storage and not hard drives.  This can make a _huge_ difference.*
+* In general try to fit the tool to the job.  Consider writing short Shell/Batch/Groovy/Python scripts when running a complex process using a build agent.  Good examples include processing data, communicating interactively with REST APIs, and parsing/templating larger XML or JSON files.  The `sh` and `bat` steps are helpful to invoke these, especially with `returnStdout: true` to return the output from this script and save it as a variable (Scripted Pipeline).
```

Comment:
Perhaps add something like *The Pipeline DSL is not designed for arbitrary networking and computation tasks* to be even clearer here? Perhaps we can convince *someone* that doing this would be dumb?